### PR TITLE
Add Badges!

### PR DESCRIPTION
**Pull request by LecrisUT**
Here are some basic badges, you can see some more in [spglib](https://github.com/spglib/spglib/blob/develop/README.md) and [shields.io](https://shields.io/badges). Other ones that I would recommend to add:
- License badge: ![License Status][license-badge]
  - This is a bit tricky if there are multiple licenses
  - Need clarification and update of the GH metadata to clarify if it is [`GPL-2.0-only`](https://spdx.org/licenses/GPL-2.0-only) or [`GPL-2.0-or-later`](https://spdx.org/licenses/GPL-2.0-or-later)
- [x] Let me know which ones you wish to promote
- Build-system badge: ![][cmake-badge]
- Compiler standard: Examples ![][fortran-badge] ![][c-badge]  ![][cpp-badge]
- CI badge. Not sure how to create this one, but if there is any json output and examples it is [possible](https://shields.io/badges/dynamic-json-badge)
- Social badge: Where the users can contact the devs.

Let me know which ones you would like added/removed and if you want them organized somehow.

[license-badge]: https://img.shields.io/github/license/cp2k/cp2k
[conda-badge]: https://img.shields.io/conda/vn/conda-forge/cp2k
[debian-badge]: https://img.shields.io/debian/v/cp2k
[cmake-badge]: https://img.shields.io/badge/CMake-3.22-blue?logo=data:image/svg%2bxml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGRhdGEtbmFtZT0iTGF5ZXIgMiIgdmlld0JveD0iMCAwIDU0NS41IDU0NS45Ij48cGF0aCBkPSJNNTQ2IDUzNCAyODIgOWwzOSA0MzQgMjI1IDkxeiIgZGF0YS1uYW1lPSJyZWQgcmlnaHQiIHN0eWxlPSJmaWxsOiNmMzI3MzU7c3Ryb2tlLXdpZHRoOjAiLz48cGF0aCBkPSJNNTI1IDU0NiAxNzAgNDAzIDEgNTQ2aDUyNHoiIGRhdGEtbmFtZT0iZ3JlZW4gYm90dG9tIiBzdHlsZT0ic3Ryb2tlLXdpZHRoOjA7ZmlsbDojM2VhZTJiIi8+PHBhdGggZD0iTTI2MyAwIDAgNTIybDI4OC0yNDRMMjYzIDB6IiBkYXRhLW5hbWU9ImJsdWUgbGVmdCIgc3R5bGU9ImZpbGw6IzAwNjhjNztzdHJva2Utd2lkdGg6MCIvPjxwYXRoIGQ9Im0yOTEgMjk5LTEwNSA4OSAxMTcgNDgtMTItMTM3eiIgZGF0YS1uYW1lPSJncmF5IGNlbnRlciIgc3R5bGU9ImZpbGw6I2RjZTNlYztzdHJva2Utd2lkdGg6MCIvPjwvc3ZnPg==
[c-badge]: https://img.shields.io/badge/C-23-blue?logo=c
[cpp-badge]: https://img.shields.io/badge/C++-20-blue?logo=cplusplus
[fortran-badge]: https://img.shields.io/badge/Fortran-08-blue?logo=fortran

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # CP2K
 
+[![Release Status][release-badge]][release-link]
+[![Fedora Status][fedora-badge]][fedora-link]
```
Comment (LecrisUT) [inline README.md]: Interesting that the pre-commit suggests compressing these lines. I don't think that's a good thing to do in this case. It creates unnecessary changes when lines are added/removed, and it is less readable.

Interesting, `mdformat` seems to be used here, which we also do on `spglib`, however, over there it doesn't suggest such changes :thinking:

Comment (oschuett) [reply README.md]: Maybe we need to [update](https://github.com/cp2k/cp2k/blob/master/tools/precommit/requirements.txt) our mdformat and perhaps also install the `mdformat-footnote` plugin? I can prepare a PR for that.

Comment (LecrisUT) [reply README.md]: Not really, it just so happens that I am hitting `--wrap=100` which is not set on `spglib`. I've made an issue upstream about it: https://github.com/executablebooks/mdformat/issues/437. I am not sure how to design it on their side, so I will probably keep this on hold and wrap according to pre-commit for now.

Comment (LecrisUT) [inline README.md]:
```suggestion
[![Release Status][release-badge]][release-link] [![Fedora Status][fedora-badge]][fedora-link]
```

**Comment by oschuett** [comment]
Thanks for adding the badges. Our top README could really use some touch up (just like our website landing page).

To avoid clutter, I'd focus on the distribution channels because these are most relevant to our users. For a complete list see also https://www.cp2k.org/download#from_a_distribution.

**Comment by LecrisUT** [comment]
- Missing distribution badges: FreeBSD, NixOS, Easybuild,  Nvidia NGC
- Questionable badges:
  - Docker: Should it be the `latest` tag?

**Comment by oschuett** [comment]
> Docker: Should it be the latest tag?

I just pushed a new tag: `2024.1` -> `2024.1_mpich_generic_psmp`. 

So, now we can use the following badge:

```
[![Docker Status](https://img.shields.io/docker/v/cp2k/cp2k?label=docker)](https://hub.docker.com/r/cp2k/cp2k)
```

[Docker Status](https://hub.docker.com/r/cp2k/cp2k)

Btw, maybe inlining yields the most readable markdown here.

**Comment by LecrisUT** [comment]
> Btw, maybe inlining yields the most readable markdown here.

Depends I guess. When you start to split it into multiple lines, or put it in a table, or do anything that is not just placing them sequentially, than it becomes harder to manage. Custom badges like the CMake one would be particularly problematic:
```markdown
 ![][cmake-badge]
[cmake-badge]: https://img.shields.io/badge/CMake-3.22-blue?logo=data:image/svg%2bxml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGRhdGEtbmFtZT0iTGF5ZXIgMiIgdmlld0JveD0iMCAwIDU0NS41IDU0NS45Ij48cGF0aCBkPSJNNTQ2IDUzNCAyODIgOWwzOSA0MzQgMjI1IDkxeiIgZGF0YS1uYW1lPSJyZWQgcmlnaHQiIHN0eWxlPSJmaWxsOiNmMzI3MzU7c3Ryb2tlLXdpZHRoOjAiLz48cGF0aCBkPSJNNTI1IDU0NiAxNzAgNDAzIDEgNTQ2aDUyNHoiIGRhdGEtbmFtZT0iZ3JlZW4gYm90dG9tIiBzdHlsZT0ic3Ryb2tlLXdpZHRoOjA7ZmlsbDojM2VhZTJiIi8+PHBhdGggZD0iTTI2MyAwIDAgNTIybDI4OC0yNDRMMjYzIDB6IiBkYXRhLW5hbWU9ImJsdWUgbGVmdCIgc3R5bGU9ImZpbGw6IzAwNjhjNztzdHJva2Utd2lkdGg6MCIvPjxwYXRoIGQ9Im0yOTEgMjk5LTEwNSA4OSAxMTcgNDgtMTItMTM3eiIgZGF0YS1uYW1lPSJncmF5IGNlbnRlciIgc3R5bGU9ImZpbGw6I2RjZTNlYztzdHJva2Utd2lkdGg6MCIvPjwvc3ZnPg==
```

**Comment by oschuett** [comment]
> Depends I guess.

Yeah, I generally try to avoid reference-style links, especially if the ref is used only once, because it adds an indirection, but you  are absolutely right that long links can mess things up quickly.
 
Could you please add `?label=docker` to the docker badge. Otherwise, LGTM.

File: README.md [hunk 2]
```diff
@@ -66,3 +72,22 @@ Additional directories created during build process:
...
 - `obj`: Objects and other intermediate compilation-time files
 - `exe`: Where the executables will be located
+
+[arch-badge]: https://img.shields.io/aur/version/cp2k
+[arch-link]: https://aur.archlinux.org/packages/cp2k
+[conda-badge]: https://img.shields.io/conda/vn/conda-forge/cp2k
+[conda-link]: https://anaconda.org/conda-forge/cp2kcp2k
+[debian-badge]: https://img.shields.io/debian/v/cp2k
+[debian-link]: https://packages.debian.org/search?keywords=cp2k
+[docker-badge]: https://img.shields.io/docker/v/cp2k/cp2k
```
Comment (LecrisUT) [inline README.md]:
```suggestion
[docker-badge]: https://img.shields.io/docker/v/cp2k/cp2k?label=docker
```

File: README.md [hunk 3]
```diff
@@ -66,3 +72,22 @@ Additional directories created during build process:
 - `lib`: Libraries built during compilation
 - `obj`: Objects and other intermediate compilation-time files
 - `exe`: Where the executables will be located
+
+[arch-badge]: https://img.shields.io/aur/version/cp2k
+[arch-link]: https://aur.archlinux.org/packages/cp2k
+[conda-badge]: https://img.shields.io/conda/vn/conda-forge/cp2k
+[conda-link]: https://anaconda.org/conda-forge/cp2kcp2k
```
Comment (oschuett) [inline README.md]:
```suggestion
[conda-link]: https://anaconda.org/conda-forge/cp2k
```
